validate/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/versionbundle"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/validation"
+	"sigs.k8s.io/yaml"
+)
+
+func scanDocuments(filename string) ([]string, error) {
+	data, err := os.Open(filename)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	scanner := bufio.NewScanner(data)
+	var document string
+	var documents []string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line != "---" {
```

Comment:
This is the kind of line that deserves a comment IMO

Reply:
Definitely agree. I'll explain with some comments.